### PR TITLE
fix(components/phone-field): phone field will switch to the correct country when a valid number for that country is given when the dial code matches multiple countries

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
[AB#2718228](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2718228)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1936/storybook/)

Component Storybooks:

- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/1936/storybooks/phone-field-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1936/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1936/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1936/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1936/playground/)

File: libs/components/phone-field/src/lib/modules/phone-field/phone-field.component.ts [hunk 1]
```diff
@@ -402,47 +402,64 @@ export class SkyPhoneFieldComponent implements OnDestroy, OnInit {
...
-            newCountry = foundCountry;
+      if (foundCountry && !this.#validateSupportedCountry(foundCountry)) {
+        foundCountry = undefined;
+      }
+    } catch (_) {
+      if (
+        !(
+          this.selectedCountry?.dialCode &&
+          phoneNumberRaw.startsWith(`${this.selectedCountry?.dialCode}`)
+        )
```
Comment (johnhwhite) [inline libs/components/phone-field/src/lib/modules/phone-field/phone-field.component.ts]: Can you simplify this for readability?
```typescript
!this.selectedCountry?.dialCode ||
!phoneNumberRaw.startsWith(`${this.selectedCountry?.dialCode}`)
```

**Review by johnhwhite — COMMENTED** [review]
One small change.

File: libs/components/phone-field/src/lib/modules/phone-field/phone-field.component.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/phone-field/src/lib/modules/phone-field/phone-field.component.ts]: Updated. I also reverted the playground change as that wasn't intended to be checked in.

**Review by johnhwhite — APPROVED** [review]

**Pull request merged**